### DISCUSSION TITLE
How to let model use more recent information when forecasting?

### DISCUSSION THREAD

**Discussion by stikket** [ourownstory/neural_prophet#1541] [Q&A - forecasting best practices] [answered]
Hi,

Is there a way for NeuralProphet to use 'y' information of more recent data?

I have sales data that has yearly, weekly, and daily seasonality. I am dealing with hourly data.

Let's say on a usual monday, at for 6 AM - $250 , 7 AM - $500 ,8 AM - $1000 sales 

Then came a monday with low sales for 6 AM and 7 AM, say 6 AM - $50 , 7 AM - $100. The model predicts 8 AM sales to be still high close to $1000. Is it possible for the model to give more importance to the past 24 hours of data available when making prediction?

**Comment by ourownstory** [comment, marked as answer]
Yes, that is possible by using Autoregression. 
Set `n_lags=24`. 
Just beware, if you were not using AR thus far with NP, this is the one setting that will affect few other settings, like you need to also explicitly define the number of `n_forecasts` steps to predict (should be less than n_lags), e.g. 12.

**Reply by stikket** [reply]
Great, have tried it and am getting better results. 

How do I interpret the yhat1,yhat2,...,yhat12 if I set to n_foreecast = 12? 
Does it mean for a specific ds, yhat1 is the forecast for that specific ds, while yhat2 is for the next time period and so on?

**Reply by ourownstory** [reply]
Great!
For a given ds, yhatX is the prediction made for this ds X steps ago.

**Reply by stikket** [reply]
Ohh... that makes more sense looking at my results... 

So if I want to look at my results in the next 24 hours, I just need to look at yhat1 of each hour? But would this take the 'y' information (it shouldn't right?)

**Reply by ourownstory** [reply]
Maybe this tutorial may be of help: https://neuralprophet.com/how-to-guides/feature-guides/collect_predictions.html
have a closer look at the 'raw' format.
